pkg/apis/core/validation/shoot.go

```diff
@@ -1743,6 +1743,32 @@ func validateCoreDNS(coreDNS *core.CoreDNS, fldPath *field.Path) field.ErrorList
 	if coreDNS.Autoscaling != nil && !availableCoreDNSAutoscalingModes.Has(string(coreDNS.Autoscaling.Mode)) {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("autoscaling").Child("mode"), coreDNS.Autoscaling.Mode, availableCoreDNSAutoscalingModes.List()))
 	}
+	if coreDNS.Rewriting != nil {
+		allErrs = append(allErrs, ValidateCoreDNSRewritingCommonSuffixes(coreDNS.Rewriting.CommonSuffixes, fldPath.Child("rewriting"))...)
+	}
+
+	return allErrs
+}
+
+// ValidateCoreDNSRewritingCommonSuffixes validates the given common suffixes used for DNS rewriting.
+func ValidateCoreDNSRewritingCommonSuffixes(commonSuffixes []string, fldPath *field.Path) field.ErrorList {
```

Comment:
You could add unit tests for the newly introduced validation.

Reply:
Addressed with latest commit.